compiler/grainformat/reformat.re

```diff
@@ -0,0 +1,2753 @@
+                      ~expr=e,
+                      ~parentIsArrow=false,
+                      ~endChar=None,
+                      ~original_source,
+                      ~parent_loc,
+                    );
+
+                  previous_line := get_end_loc_line(e.pexp_loc);
+
+                  Doc.group(
+                    Doc.concat([
+                      stmt_leading_comment_docs,
+                      blank_line_above,
+                      Doc.group(printed_expression),
+                    ]),
+                  );
+                };
+              },
+              expressions,
+            ),
+          );
+
+        Doc.breakableGroup(
+          ~forceBreak=true,
+          Doc.concat([
+            Doc.lbrace,
+            Doc.indent(Doc.concat([Doc.line, block])),
+            Doc.line,
+            Doc.rbrace,
+          ]),
+        );
+      } else {
+        let (_leading_comments, trailing_comments) =
+          Walktree.partition_comments(expr.pexp_loc, Some(parent_loc));
+
+        Walktree.remove_used_comments([], trailing_comments);
+
+        let expr_line = get_end_loc_line(expr.pexp_loc) + 1;
+
+        let blockCommentDocs =
+          if (List.length(trailing_comments) > 0) {
+            Doc.concat([
+              print_multi_comments_no_space(trailing_comments, expr_line),
+              Doc.hardLine,
+            ]);
+          } else {
+            Doc.nil;
+          };
+        Doc.breakableGroup(
+          ~forceBreak=true,
+          Doc.concat([
+            Doc.lbrace,
+            Doc.indent(Doc.concat([Doc.line, blockCommentDocs])),
+            Doc.line,
+            Doc.rbrace,
+          ]),
+        );
+      }
+
+    | PExpBoxAssign(expression, expression1) =>
+      Doc.concat([
+        print_expression(
+          ~expr=expression,
+          ~parentIsArrow=false,
+          ~endChar=None,
+          ~original_source,
+          ~parent_loc,
+        ),
+        Doc.text(" := "),
+        print_expression(
+          ~expr=expression1,
+          ~parentIsArrow=false,
+          ~endChar=None,
+          ~original_source,
+          ~parent_loc,
+        ),
+      ])
+    | PExpAssign(expression, expression1) =>
+      switch (expression1.pexp_desc) {
+      | PExpApp(func, expressions) =>
+        let functionName = get_function_name(func);
+
+        let trimmed_operator = String.trim(functionName);
+
+        let left =
+          print_expression(
+            ~expr=expression,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~parent_loc,
+            ~original_source,
+          );
+
+        let first =
+          print_expression(
+            ~expr=List.hd(expressions),
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~parent_loc,
+            ~original_source,
+          );
+
+        if (left == first) {
+          // +=, -=, *=, /=, and %=
+          switch (trimmed_operator) {
+          | "+"
```

Comment:
An infix list would also be really good.

Reply:
Hey @ospencer   not sure what you mean here.  I have an infix list but I thought these were the only ones sugared.  Should I make a sublist?

Reply:
Thanks for the conversation, all good.